googletest/include/gtest/internal/gtest-type-util.h

```diff
@@ -85,6 +101,19 @@ inline std::string GetTypeName(const std::type_info& type) {
   const std::string name_str(status == 0 ? readable_name : name);
   free(readable_name);
   return CanonicalizeForStdLibVersioning(name_str);
+#elif defined(_MSC_VER)
+  // Strip struct and class due to differences between
+  // MSVC and other compilers. std::pair<int,bool> is printed as
+  // "struct std::pair<int,bool>" when using MSVC vs "std::pair<int, bool>" with
+  // other compilers.
+  std::string s = name;
+  if (s.rfind("struct ", 0) == 0) {
```

Comment:
Why not just find ?

Reply:
Only strip the leading "struct " and "class ", so uses rfind == 0 to ensure that